### PR TITLE
Support specifying range of versions.

### LINKED ISSUE #74: Support exact_version
`pkg-config` provides `name = version` syntax to specify the exact version. It would be nice to add a wrapper for this functionality.  Although `pkg_config::probe_library("name = version")` works, I think it is abuse.

### PR DESCRIPTION
**Pull request by Hakuyume**
close #74 

This PR adds two new methods: `eq_version` and `range_version`.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -283,7 +286,33 @@ impl Config {
...
     /// Indicate that the library must be at least version `vers`.
     pub fn atleast_version(&mut self, vers: &str) -> &mut Config {
-        self.atleast_version = Some(vers.to_string());
+        self.min_version = Bound::Included(vers.to_string());
+        self.max_version = Bound::Unbounded;
+        self
+    }
+
+    /// Indicate that the library must be equal to version `vers`.
+    pub fn eq_version(&mut self, vers: &str) -> &mut Config {
```
Comment (sdroege) [inline src/lib.rs]: Maybe `exactly_version()`?

File: src/lib.rs [hunk 2]
```diff
@@ -283,7 +286,33 @@ impl Config {
...
+        self.min_version = Bound::Included(vers.to_string());
+        self.max_version = Bound::Included(vers.to_string());
+        self
+    }
+
+    /// Indicate that the library's version must be in `range`.
+    pub fn range_version<'a, R>(&mut self, range: R) -> &mut Config
+    where
+        R: RangeBounds<&'a str>
```
Comment (sdroege) [inline src/lib.rs]: With the `RangeBounds` is a great idea :)

File: tests/test.rs [hunk 3]
```diff
@@ -116,3 +116,85 @@ fn version() {
...
+    reset();
+    pkg_config::Config::new().range_version("3.10".."3.12").probe("foo").unwrap();
+}
+
+#[test]
+#[should_panic]
+fn range_version_range_ng() {
+    let _g = LOCK.lock();
+    reset();
+    pkg_config::Config::new().range_version("3.12".."3.15").probe("foo").unwrap();
```
Comment (sdroege) [inline tests/test.rs]: Maybe also add tests for inclusive vs. exclusive ranges

**Comment by sdroege** [comment]
Great, looks good to me. @alexcrichton, good for you too?

**Comment by alexcrichton** [comment]
Indeed!

**Pull request merged**